gazebo_ros_control/include/gazebo_ros_control/gazebo_ros_control_plugin.h

```diff
@@ -0,0 +1,135 @@
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Dave Coleman, Jonathan Bohren
+   Desc:   Gazebo plugin for ros_control that allows 'hardware_interfaces' to be plugged in
+           using pluginlib
+*/
+
+// Boost
+#include <boost/shared_ptr.hpp>
+#include <boost/thread.hpp>
+
+// ROS
+#include <rclcpp/rclcpp.hpp>
+#include <pluginlib/class_loader.hpp>
+#include <std_msgs/msg/bool.hpp>
+
+// Gazebo
+#include <gazebo/gazebo.hh>
+#include <gazebo/physics/physics.hh>
+#include <gazebo/common/common.hh>
+
+// ros_control
+#include <gazebo_ros_control/robot_hw_sim.h>
+#include <controller_manager/controller_manager.hpp>
+#include <transmission_interface/transmission_parser.hpp>
+
+namespace gazebo_ros_control
+{
+
+class GazeboRosControlPlugin : public gazebo::ModelPlugin
+{
+public:
+
+  virtual ~GazeboRosControlPlugin();
+
+  // Overloaded Gazebo entry point
+  virtual void Load(gazebo::physics::ModelPtr parent, sdf::ElementPtr sdf);
+
+  // Called by the world update start event
+  void Update();
+
+  // Called on world reset
+  virtual void Reset();
+
+  // Get the URDF XML from the parameter server
+  std::string getURDF(std::string param_name) const;
+
+  // Get Transmissions from the URDF
+  bool parseTransmissionsFromURDF(const std::string& urdf_string);
+
+protected:
+  void eStopCB(const std::shared_ptr<std_msgs::msg::Bool> e_stop_active);
+
+  // Node Handles
+  rclcpp::Node::SharedPtr model_nh_; // namespaces to robot name
+
+  // Pointer to the model
+  gazebo::physics::ModelPtr parent_model_;
+  sdf::ElementPtr sdf_;
+
+  // deferred load in case ros is blocking
+  boost::thread deferred_load_thread_;
+
+  // Pointer to the update event connection
+  gazebo::event::ConnectionPtr update_connection_;
+
+  // Interface loader
+  boost::shared_ptr<pluginlib::ClassLoader<gazebo_ros_control::RobotHWSim> > robot_hw_sim_loader_;
+  void load_robot_hw_sim_srv();
+
+  // Strings
+  std::string robot_namespace_;
+  std::string robot_description_;
+
+  // Transmissions in this plugin's scope
+  std::vector<transmission_interface::TransmissionInfo> transmissions_;
+
+  // Robot simulator interface
+  std::string robot_hw_sim_type_str_;
+  std::shared_ptr<gazebo_ros_control::RobotHWSim> robot_hw_sim_;
+  rclcpp::executors::MultiThreadedExecutor::SharedPtr executor_;
+  std::thread thread_executor_spin_; //executor_->spin causes lockups, us ethis alternative for now
+
+  // Controller manager
+  std::shared_ptr<controller_manager::ControllerManager> controller_manager_;
+  std::shared_ptr<controller_interface::ControllerInterface> controller_;
+  std::shared_ptr<controller_interface::ControllerInterface> controller2_;
```

Comment:
These 2 controllers are custom code. Normally you'd get the controller_manager to spawn controllers but this functionality is not available as of this writing.